internal/impl/nats/metadata.go

```diff
@@ -26,3 +26,19 @@ func newMessageFromKVEntry(entry jetstream.KeyValueEntry) *service.Message {
 
 	return msg
 }
+
+func newMessageWithMetaFromKVEntry(entry jetstream.KeyValueEntry, src *service.Message) *service.Message {
+	msg := service.NewMessage(entry.Value())
+	//nolint:errcheck
+	src.MetaWalkMut(func(key string, value any) error {
+		msg.MetaSetMut(key, value)
+		return nil
+	})
```

Comment:
Would probably be easier to just set the bytes of the original message to the new content. That way we can retain all the original metadata without needing to construct a new message from the KV entry.

```suggestion
	msg := src.Copy()
	msg.SetBytes(entry.Value())
```

This way would also retain the original message's context -- and therefore its tracing ID.